src/de.rs

```diff
@@ -348,19 +366,28 @@ impl<'de> de::Deserializer<'de> for Deserializer {
         }
     }
 
-    // Define real `i64` / `u64` deserializers that try to cast from `f64`.
-
     fn deserialize_i64<V: de::Visitor<'de>>(self, visitor: V) -> Result<V::Value> {
-        match self.as_safe_integer() {
-            Some(v) => visitor.visit_i64(v),
-            None => self.invalid_type(visitor),
+        if self.value.is_bigint() {
+            let big_int: BigInt = self.value.clone().into();
+            // Safe to do unwraps since we know the type is a bigint
+            let rust_string = big_int.to_string(10).unwrap().as_string().unwrap();
```

Comment:
This one is still using strings, but shouldn't anymore.

Reply:
oops